### DISCUSSION TITLE
State for last update time

### DISCUSSION THREAD

**Discussion by shuding** [vercel/swr#2319] [Ideas]
It would be helpful to introduce a new state `updatedAt`, which is a timestamp representing the last time we get the data from the fetcher or mutator.

## Use Case

The typical use case for this feature will be showing a last refreshed time of the data in the UI which is critical for its realtimeness.

Without this new state one can only implement it with an effect hook to monitor the data change. However due to the deep comparison of SWR that doesn't work when data remains the same.

**Comment by icyJoseph** [comment]
`updatedAt` then should be equal to `Date.now` right after the concurrent promise for this key resolves?

```ts
      startAt = CONCURRENT_PROMISES_TS[key]
      newData = await CONCURRENT_PROMISES[key]
      newState.updatedAt = Date.now()
```

Of course keeping the `updatedAt` as part of the tracked dependencies state.

And is anyone actively working on this? 

Edit: I now have some local changes with this change.
